spec.bs

```diff
@@ -441,13 +439,30 @@ partial dictionary RequestInit {
 Modifications to request {#fetch-request}
 ---------------------------------------------------------------
 
-A [=/request=] has an associated <dfn for=request>token refresh policy</dfn>, which is <code>"none"</code> or <code>"refresh"</code>. Unless stated otherwise it is <code>"none"</code>.
+A [=/request=] has an associated <dfn for=request>token refresh policy</dfn>, which is of type {{RefreshPolicy}} with default value of <code>"none"</code>.  [=request/Token refresh policy=] specifies whether to <a lt="set redemption headers">execute redemption procedure</a> or [=retrieve a redemption record=].
 
-A [=request=] has an associated <dfn for="request">pstPretokens</dfn>, which is null or a [=byte sequence=].
+A [=/request=] has an associated <dfn for=request>operation</dfn>, which is of type {{OperationType}}. [=request/operation=] specifies the Private Token operation.
+
+A [=/request=] has an associated <dfn for=request>issuers</dfn>, which is a list of strings.
 
-Add the following steps to the <code><a constructor lt="Request()">new Request (<var ignore>input</var>, |init|])</a></code> constructor, before step 28 ("<code>Set [=this=]'s [=Request/request=] to |request|</code>"):
+[=request/Token refresh policy=] is ignored unless [=request/operation=] is <code>"token-redemption"</code>. [=request/issuers=] is ignored unless [=request/operation=] is <code>"send-redemption-record"</code>. [=request/issuers=] must be specified and non-empty when [=request/operation=] is <code>"send-redemption-record"</code>.
+
+A [=request=] has an associated <dfn for="request">pstPretokens</dfn>, which is null or a [=byte sequence=].
 
- 1. If |init|["{{RequestInit/privateToken}}"] [=map/exists=], then set <var ignore>request</var>'s [=request/token refresh policy=]</a> to |init|["{{RequestInit/privateToken}}"]["{{PrivateToken/refreshPolicy}}"].
+Add the following steps to the <code><a constructor lt="Request()">new Request (<var ignore>input</var>, |init|)</a></code> constructor, before step 28 ("<code>Set [=this=]'s [=Request/request=] to |request|</code>"):
+
+ 1. If |init|["{{RequestInit/privateToken}}"] [=map/exists=]:
+     1. Set <var ignore>request</var>'s [=request/operation=]</a> to |init|["{{RequestInit/privateToken}}"]["{{PrivateToken/operation}}"].
```

Comment:
You're not using the operation right now. The operation should be used in e.g. `Modifications to http-network-or-cache fetch` to determine which algorithm to run, if any.

I'm slightly confused that there are 3 operation types but only 2 algorithms to run in the fetch steps, but I'm probably missing something here.

Reply:
Added algorithm for sending redemption records. Added a new step to `Modifications to http-network-or-cache fetch`.